easynav_gps_vff/easynav_gps_vff_controller/src/easynav_gps_vff_controller/GpsVffController.cpp

```diff
@@ -0,0 +1,306 @@
+// Copyright 2025 Intelligent Robotics Lab
+//
+// This file is part of the project Easy Navigation (EasyNav in short)
+// licensed under the GNU General Public License v3.0.
+// See <http://www.gnu.org/licenses/> for details.
+//
+// Easy Navigation program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+/// \file
+/// \brief Implementation of the GpsVffController class.
+
+#include <expected>
+
+#include "easynav_gps_vff_controller/GpsVffController.hpp"
+
+namespace easynav
+{
+
+std::expected<void, std::string> GpsVffController::on_initialize()
+{
+  auto node = get_node();
+  const auto & plugin_name = get_plugin_name();
+
+  node->declare_parameter<float>(plugin_name + ".distance_obstacle_detection", 3.0);
+  node->declare_parameter<float>(plugin_name + ".distance_to_goal", 1.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_x_min", 0.5);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_x_max", 10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_y_min", -10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_y_max", 10.0);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_z_min", 0.10);
+  node->declare_parameter<float>(plugin_name + ".obstacle_detection_z_max", 1.00);
+  node->declare_parameter<double>(plugin_name + ".max_speed", 0.8);
```

Comment:
Maybe, in a future improved version, we should also incorporate the acceleration, and `max_speed` will be reached by increasing the current velocity using the time and the acceleration.